challenger/lnd.go

```diff
@@ -84,49 +89,61 @@ func (l *LndChallenger) Start() error {
 	addIndex := uint64(0)
 	settleIndex := uint64(0)
 
-	// Get a list of all existing invoices on startup and add them to our
-	// cache. We need to keep track of all invoices, even quite old ones to
-	// make sure tokens are valid. But to save space we only keep track of
-	// an invoice's state.
+	// Paginate through all existing invoices on startup and add them to our
+	// cache. We need to keep track of all invoices to ensure tokens are
+	// valid.
 	ctx := l.clientCtx()
-	invoiceResp, err := l.client.ListInvoices(
-		ctx, &lnrpc.ListInvoiceRequest{
-			NumMaxInvoices: math.MaxUint64,
-		},
-	)
-	if err != nil {
-		return err
-	}
-
-	// Advance our indices to the latest known one so we'll only receive
-	// updates for new invoices and/or newly settled invoices.
-	l.invoicesMtx.Lock()
-	for _, invoice := range invoiceResp.Invoices {
-		// Some invoices like AMP invoices may not have a payment hash
-		// populated.
-		if invoice.RHash == nil {
-			continue
+	indexOffset := uint64(0)
+	for {
+		invoiceResp, err := l.client.ListInvoices(
```

Comment:
Could also add some debug logging here so we could observe the startup progress in the logs.